pubtools/pulplib/_impl/model/repository/base.py

```diff
@@ -595,6 +597,30 @@ def sync(self, options=None):
             )
         )
 
+    def lock(self, duration=LOCK_EXPIRATION_OFFSET, retry_wait=AWAIT_ACTIVATION_SLEEP):
+        """
+        Lock the repo to show other tasks that it is being worked on.
+        This does not prevent modifications to the repo with the Pulp API, it
+        just provides a way to pause tasks to prevent race conditions.
+
+        Args:
+            duration
+                How long the lock should remain valid for in seconds.
+            retry_wait:
+                How long to wait between attempting to activate the lock in
+                seconds.
+        """
+        return RepoLock(
+            self.id,
+            lambda url, **kwargs: self._client._request_executor.submit(
+                self._client._do_request,
+                url=os.path.join(self._client._url, url),
+                **kwargs
+            ),
+            expiration_offset=duration,
```

Comment:
I'd like to request that we clamp the caller's input to a reasonable range somewhere in here, because of the high impact in case the user provides bad values.  e.g.

- if the caller passes a too low duration, we can end up with the same silent data corruption we're trying to fix here.
- if the caller passes a too high duration, we can end up with extreme delays/apparent deadlock in the cases where a stale lock needs to be released.

I'll make a rough guess:

- if caller requested a duration less than 5 minutes, we should force it to 5 minutes (because basically anything could take that long if there's an outage which forces requests to be retried).
- if caller requested a duration more than 6 hours, we should force it to 6 hours (kind of arbitrarily picked as one quarter of a day and fitting into one working day... and nothing comes to mind which should ever genuinely need the lock for longer than that.)

I prefer "clamp" rather than "validate" because if the approach is to validate, then later changes to it are potentially backwards-incompatible (e.g. if we accept 5 minutes today but later need to change it to 15 minutes, that could break someone).